komposable-architecture-test/src/main/java/com/toggl/komposable/test/store/TestStoreAssertionRunner.kt

```diff
@@ -0,0 +1,192 @@
+package com.toggl.komposable.test.store
+
+import com.toggl.komposable.test.utils.info
+import io.kotest.matchers.shouldBe
+import kotlin.reflect.full.memberProperties
+
+internal fun <State : Any, Action> TestStore.Exhaustivity.createAssertionRunner(
+    testStore: TestStore<State, Action>,
+): TestStore.AssertionRunner<State, Action> =
+    if (this is TestStore.Exhaustivity.NonExhaustive) {
+        NonExhaustiveAssertionRunner(
+            testStore,
+            this.logIgnoredStateChanges,
+            this.logIgnoredReceivedActions,
+            this.logIgnoredEffects,
+        )
+    } else {
+        ExhaustiveAssertionRunner(testStore)
+    }
+
+internal class ExhaustiveAssertionRunner<State : Any, Action>(
+    store: TestStore<State, Action>,
+) : TestStore.AssertionRunner<State, Action> {
+    private val reducer = store.reducer
+    private val timeout = store.timeout
+    override fun assertStateChange(
+        previousState: State,
+        currentState: State,
+        assert: ((state: State) -> State)?,
+    ) {
+        if (assert != null) {
+            currentState shouldBe assert(previousState)
+        } else {
+            currentState shouldBe previousState
+        }
+    }
+
+    override fun hasReceivedActionToHandle(match: (Action) -> Boolean): Boolean =
+        reducer.receivedActions.isNotEmpty()
```

Comment:
Do we want to use the `match` function here? I think ExhaustiveAssertionRunner should verify that only 1 action received and it matches the predicate?

Reply:
We are not using it in the `ExhaustiveAssertionRunner` because we don't need to know which action has been received at the point 'this gets called, only that A action has been received. The exhaustive behaviour expects actions to be received in order, so if any action has been received, it MUST be the one the user is expecting or the test should fail. The actual matching happens in `receiveAction`.